src/connection/postgres_connection_factory.py

```
@@ -0,0 +1,47 @@
+class PostgresConnectionFactory(ConnectionFactory):
+    def __init__(self) -> None:
+        super().__init__()
+    
+    def read_file(self, conn_id: str, file_path:str) -> dict[str, str]:
+        data = None
+        
+        try:
+            with open(file_path) as file:
+                data = json.load(file)
+        except IOError as e:
+            raise IOError(f"Falha ao ler arquivo:\n{e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)